Autopilot/Inc/PID.hpp

```diff
@@ -0,0 +1,56 @@
+/**
+ * @file PID.hpp
+ * @author Ian Frosst
+ * @date  March 26, 2017
+ * @copyright Waterloo Aerial Robotics Group 2017 \n
+ *  https://raw.githubusercontent.com/UWARG/ZeroPilot-SW/devel/LICENSE.md
+ */
+
+#ifndef PID_HPP
+#define	PID_HPP
+
+#include <cmath>
+#include <cstdint>
+
+/***********************************************************************************************************************
+ * Prototypes
+ **********************************************************************************************************************/
+
+class PIDController
+{
+    public:
+    	/**
+		* Initialises the Pid object.
+		* @param[in]	_kp 	The proportional gain.
+		* @param[in]	_ki 	The integral gain.
+		* @param[in]	_kd  	The derivative gain.
+		* @param[in]	_i_max	Max value the integral should ever be allowed to take on.
+		* @param[in]	_min_output		The minimum value that can be output, if computations return smaller, the output will be set to this value.
+		* @param[in]	_max_output		The maximum value that can be output, if computations return larger, the output will be set to this value.
+		*/
+        PIDController(float _kp, float _ki, float _kd, float _i_max, float _min_output, float _max_output);
+
+    	/**
+		* Executes a PID computation.
+		* The PID algorithm uses the derivative of the actual values, rather than the derivatives of the error. That makes it immune to sudden changes in commanded set points.
+		* THIS METHOD MUST BE CALLED AT A REGULAR INTERVAL!!! UNDEFINED BEHAVIOUR WILL OCCUR OTHERWISE.
+		* Will use the measured derivative if provided with one. Otherwise, will compute the derivative from the given measurements.
+		* @param[in]	desired 	The point we wish to reach.
+		* @param[in]	actual 		The current point.
+		* @param[in]	actualRate  The current measured derivative (This parameter is optional).
+		* @return					The result of the PID computation.
+		*/
+        float execute(float desired, float actual, float actualRate = std::nanf(""));
+
+    private:
+
+		float kp, kd, ki;
+		int16_t i_max;
+		float integral;
```

Comment:
Not really an issue tbh, but the indentations could be aligned. Makes the code more readable.

Reply:
true. This has to do with the way GitHub displays tabs vs spaces. That file had a mix of both, but you're right. I'll just use one or the other so the indentations are consistent on any editor.

Reply:
I didn't know GitHub did that. I'll be sure to keep an eye out for it too! :))